### DISCUSSION TITLE
How to handle knowing transcription is complete when live streaming?

### DISCUSSION THREAD

**Discussion by sgodin** [deepgram/community#933] [General help]
We are using live streaming to transcribe phone calls.  We are using the C# client API, however that is probably irrelevant to my question.

We are currently initializing with the following settings:

>             var liveSchema = new LiveSchema()
>             {
>                 Model = "nova-2-phonecall",
>                 Version = "latest",
>                 Language = "en-US",
>                 Encoding = "linear16", 
>                 SampleRate = 8000, 
>                 Channels = 1, 
>                 MultiChannel = false,
>                 SmartFormat = true,
>                 InterimResults = true,
>                 VadEvents = true,
>                 UtteranceEnd = 2000,
>                 Punctuate = false, 
>                 FillerWords = false,
>                 EndPointing = 2000
>             };

When the phone call is done we send a Finalize request on the websocket.  Then we wait for transcription to be finished by looking for a result with FromFinalize set to true.

Most of the time I get back a final result with FromFinalize set to true as expected.

The documentation seems to indicate the FromFinalize set to true MAY not happen, see here:  https://developers.deepgram.com/docs/finalize
"In most cases, you will receive this response, but it is not guaranteed if there is no significant amount of audio data to process."

I am seeing the odd case where I do NOT get a FromFinalize response.  **In this case how am I supposed to know that the transcription process is actually done?**

The only solution I can think of is to start my own timer after calling Finalize and just assume things are done if I don't get a response with FromFinalize set to true, however I fear I might be firing it too early and there could actually be a result that would have arrived.

What is the recommended approach here?

Best Regards,
Scott

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by Dhruvil1726** [comment]
I have the same question !

**Comment by sgodin** [comment]
FYI - I ended up sending a CloseStream request.  Using custom code, instead of calling SendClose() to ensure all audio is processed first.

                    byte[] data = Encoding.ASCII.GetBytes("{\"type\": \"CloseStream\"}");
                    activeRecognition.SpeechClient.SendMessage(data);  // queued to ensure all audio get's there first

The server will then terminate the websocket after the final response is sent, so I look for a CloseResponse to know when done.
